internal/controller/atlasproject/custom_roles.go

```diff
@@ -50,16 +50,27 @@ func getLastAppliedCustomRoles(atlasProject *akov2.AtlasProject) ([]akov2.Custom
 	return lastAppliedSpec.CustomRoles, nil
 }
 
-func findRolesToDelete(prevSpec, atlasRoles []customroles.CustomRole) map[string]customroles.CustomRole {
+func findRolesToDelete(prevSpec, akoRoles, atlasRoles []customroles.CustomRole) map[string]customroles.CustomRole {
 	result := map[string]customroles.CustomRole{}
-	for atlasRoleIdx := range atlasRoles {
-		for specRoleIdx := range prevSpec {
-			if atlasRoles[atlasRoleIdx].Name == prevSpec[specRoleIdx].Name {
-				result[prevSpec[specRoleIdx].Name] = prevSpec[specRoleIdx]
-				continue
-			}
+	combinedAkoRoles := map[string]customroles.CustomRole{}
```

Comment:
The name `combinedAkoRoles` is confusing as it has no semantic context, what does "combined" mean?

For a future reader `deletionCandidates` reads more reasonable. I had to draw down the maps down to understand the maning of the `combinedAkoRoles`, realizing that these are the candidates to be deleted if they are present in Atlas.

Reply:
`combinedAkoRoles` means combined from the spec and the last applied config, not `deletionCandidates`. Deletion candidates are stored in the "result" map.